sklearn/svm/base.py

```diff
@@ -482,8 +482,8 @@ def _validate_for_predict(self, X):
     @property
     def coef_(self):
         if self.kernel != 'linear':
-            raise ValueError('coef_ is only available when using a '
-                             'linear kernel')
+            raise AttributeError('coef_ is only available when using a '
```

Comment:
This is one of the potentially controversial change.

From the comment in https://github.com/scikit-learn/scikit-learn/pull/3937#r21423927:

> The exception is where it would be breaking backwards compatibility to, for example, change an existing error from a ValueError to an AttributeError (although perhaps we could define a custom BackwardsCompatibleAttributeError that extends ValueError and AttributeError!).

Should we do that for this one case? Also just curious, if we do that is there a clever deprecation mechanism that allows to turn it into an AttributeError after two major releases?